src/Forms/GridField/GridFieldAddExistingAutocompleter.php

```diff
@@ -349,7 +350,18 @@ public function scaffoldSearchFields($dataClass)
                     if (!$customSearchableFields || array_search($name, $customSearchableFields)) {
                         $filter = 'StartsWith';
                     } else {
-                        $filter = preg_replace('/Filter$/', '', $spec['filter']);
+                        $filterName = $spec['filter'];
+                        // It can be an instance
+                        if ($filterName instanceof SearchFilter) {
+                            $filterName = get_class($filterName);
+                        }
+                        // It can be a fully qualified class name
+                        if (strpos($filterName, '\\') !== false) {
```

Comment:
Can we use `if (ClassInfo::exists($filterName)) {` here? If the class wasn't namespace'd the current suggested check would fail.

Reply:
@michalkleiner there are many ways this could be improved. My PR is just an improvement over the current (broken) solution. I'm specifically checking for \ for classes since it's probably not recommended to use plain class names which could be confused with aliases. As in most of my PR, if the team thinks it's necessary to make changes / add unit testing / etc they are more than welcome to do it, but I don't have time to do more than what i do (which is providing some bits of code that work in my projects and might help others as well)

Also I think it would be really nice to pick some of the changes I've made in here https://github.com/lekoala/silverstripe-base/blob/db64a738ae206ab03bda4b81b810f4f0e0de0794/src/Forms/GridField/BetterGridFieldAddExistingAutocompleter.php . The current autocompleter is just too restrictive in the way it works at the moment.